azure_storage/azure_storage_api.py

```diff
@@ -234,3 +236,77 @@ async def get_directories(container_client):
     except FolderListError as error:
         print(error)
         return []
+
+async def get_pipeline_info(
+        connection_string: str,
+        pipeline_container_name: str,
+        pipeline_version: str
+    ) -> json:
+    """
+    Retrieves the pipeline information from Azure Blob Storage based on the provided parameters.
+
+    Args:
+        connection_string (str): The connection string for the Azure Blob Storage.
+        pipeline_container_name (str): The name of the container where the pipeline files are stored.
+        pipeline_version (str): The version of the pipeline to retrieve.
+
+    Returns:
+        json: The pipeline information in JSON format.
+
+    Raises:
+        PipelineNotFoundError: If the specified version of the pipeline is not found.
+    """
+    try:
+        blob_service_client = BlobServiceClient.from_connection_string(
+            connection_string
+        )
+
+        if blob_service_client is None:
+            raise PipelineNotFoundError("No Blob Service Client found with the connection string.")
+
+        container_client = blob_service_client.get_container_client(
+            pipeline_container_name
+        )
+
+        blob = await get_blob(container_client, f"pipelines/{pipeline_version}.json")
+        pipeline = json.loads(blob)
+        return pipeline
+
+    except (ValueError, GetBlobError, PipelineNotFoundError) as error:
+        raise PipelineNotFoundError(f"This version {pipeline_version} was not found") from error
+
+
+def insert_new_version_pipeline(
+        pipelines_json: dict,
+        connection_string: str,
+        pipeline_container_name: str
+    ) -> bool:
+    """
+    Inserts a new version of a pipeline JSON into an Azure Blob Storage container.
+
+    Args:
+        pipelines_json (dict): The JSON data of the pipeline.
+        connection_string (str): The connection string for the Azure Blob Storage account.
+        pipleine_container_name (str): The name of the container where the pipeline JSON will be uploaded.
+
+    Returns:
+        bool: True if the pipeline JSON was successfully uploaded, False otherwise.
+    """
+    try:
+        blob_service_client = BlobServiceClient.from_connection_string(
+            connection_string
+        )
+
+        if blob_service_client:
+            container_client = blob_service_client.get_container_client(
+                pipeline_container_name
+            )
+
+            json_name = "{}/{}.json".format("pipelines", pipelines_json.get("version"))
+            container_client.upload_blob(json_name, json.dumps(pipelines_json, indent=4), overwrite=True)
+            return True
+        else:
+            raise ConnectionStringError("Invalid connection string")
+    except ConnectionStringError as error:
+        print(error)
+        return False
```

Comment:
best to raise an exception as first of all, we'd want why the function fails (```from``` keyword), this is by definition an exceptional situation and it is easy for the caller to ignore a boolean return and "ass-u-me" that the function will always work (until it doesn't). the caller however cannot ignore an exception.

Reply:
This is the subject of issue #61 

I have changed this function, but a lot of functions in the backend return False when catching an exception.